<a href="https://colab.research.google.com/github/AlvaroBecker/Proyecto-2/blob/parte1/parte1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Proyecto 2 - Parte 1 (base)[DE1] 

El segundo proyecto tendrá más libertad que el primero. Esto es porque queremos que tengan un proyecto en sus portafolios que les interese o se relacione con la industria en la que quieren trabajar.

La tarea para esta semana es proponer dos posibles conjuntos de datos que les gustaría trabajar para el proyecto 2.  

Escogerán el conjunto de datos de primera opción, y un conjunto de datos de reserva en caso de que el primero propuesto no se apruebe.  

Estos datos pueden proceder de cualquier fuente y ser sobre cualquier tema con estas limitaciones:

los datos deben estar disponibles para su uso (es su responsabilidad garantizar que la licencia indique que pueden utilizarlos)

los datos deben ser adecuados para un ambiente profesional

los datos NO deben contener información personal

los datos NO deben ser de un conjunto de datos usados en alguna tarea o clase del curso

Asegúrense de seleccionar un conjunto de datos con el que sea razonable trabajar en el tiempo que nos queda. Piensen qué preguntas podrían responder con el conjunto de datos que seleccionen. 

Deben proponer dos conjuntos de datos que tengan cada uno un componente de aprendizaje supervisado. Pueden elegir un problema de regresión o clasificación para cada conjunto de datos propuesto.  

Para esta tarea:

Creen un cuaderno de Colab donde hayan subido y mostrado el .head() de cada uno de los conjuntos de datos.  Para cada uno de los conjuntos de datos propuestos, respondan las siguientes preguntas:

Primera opción: conjunto de datos 1

1) Fuente de datos

2) Breve descripción de datos

3) ¿Cuál es el objetivo?

4) ¿Este es un problema de clasificación o regresión?

5) ¿Cuántas características hay?

6) ¿Cuántas filas de datos hay?

7) ¿Qué retos, si es que hay alguno, prevén en la limpieza, exploración o modelización con este conjunto de datos?

Segunda opción: conjunto de datos 2

1) Fuente de datos

2) Breve descripción de datos

3) ¿Cuál es el objetivo?

4) ¿Este es un problema de clasificación o regresión?

5) ¿Cuántas características hay?

6) ¿Cuántas filas de datos hay?

7) ¿Qué retos, si es que hay alguno, prevén en la limpieza, exploración o modelización con este conjunto de datos?

Agreguen el cuaderno de Colab a Github y envíen el enlace abajo.


In [12]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [13]:
import pandas as pd

In [14]:
df_atlas=pd.read_csv('/content/drive/MyDrive/Coding_dojo/Proyecto2/atlas-higgs-challenge-2014-v2.csv')

https://opendata.cern.ch/record/328 Dataset from the ATLAS Higgs Boson Machine Learning Challenge 2014

| Variable                      | Description                                                                                                      |
|-------------------------------|------------------------------------------------------------------------------------------------------------------|
| EventId                       | An unique integer identifier of the event.                                                                       |
| DER_mass_MMC                  | The estimated mass mH of the Higgs boson candidate, obtained through a probabilistic phase space integration.    |
| DER_mass_transverse_met_lep   | The transverse mass between the missing transverse energy and the lepton.                                       |
| DER_mass_vis                  | The invariant mass of the hadronic tau and the lepton.                                                          |
| DER_pt_h                      | The modulus of the vector sum of the transverse momentum of the hadronic tau, the lepton and the missing transverse energy vector.|
| DER_deltaeta_jet_jet          | The absolute value of the pseudorapidity separation between the two jets (undefined if PRI_jet_num ≤ 1).       |
| DER_mass_jet_jet              | The invariant mass of the two jets (undefined if PRI_jet_num ≤ 1).                                              |
| DER_prodeta_jet_jet           | The product of the pseudorapidities of the two jets (undefined if PRI_jet_num ≤ 1).                             |
| DER_deltar_tau_lep            | The R separation between the hadronic tau and the lepton.                                                       |
| DER_pt_tot                    | The modulus of the vector sum of the missing transverse momenta and the transverse momenta of the hadronic tau, the lepton, the leading jet (if PRI_jet_num ≥ 1) and the subleading jet (if PRI jet num = 2) (but not of any additional jets).|
| DER_sum_pt                    | The sum of the moduli of the transverse momenta of the hadronic tau, the lepton, the leading jet (if PRI jet num ≥ 1) and the subleading jet (if PRI jet num = 2) and the other jets (if PRI jet num = 3).|
| DER_pt_ratio_lep_tau          | The ratio of the transverse momenta of the lepton and the hadronic tau.                                         |
| DER_met_phi_centrality        | The centrality of the azimuthal angle of the missing transverse energy vector w.r.t. the hadronic tau and the lepton.|
| DER_lep_eta_centrality        | The centrality of the pseudorapidity of the lepton w.r.t. the two jets (undefined if PRI_jet_num ≤ 1).         |
| PRI_tau_pt                    | The transverse momentum p2x+p2y−−−−−−√ of the hadronic tau.                                                     |
| PRI_tau_eta                   | The pseudorapidity η of the hadronic tau.                                                                       |
| PRI_tau_phi                   | The azimuth angle ϕ of the hadronic tau.                                                                        |
| PRI_lep_pt                    | The transverse momentum p2x+p2y−−−−−−√ of the lepton (electron or muon).                                       |
| PRI_lep_eta                   | The pseudorapidity η of the lepton.                                                                             |
| PRI_lep_phi                   | The azimuth angle ϕ of the lepton.                                                                              |
| PRI_met                        | The missing transverse energy $\overrightarrow{E}^{miss}_{T}$                                                    |
| PRI_met_phi                   | The azimuth angle ϕ of the missing transverse energy                                                             |
| PRI_met_sumet                 | The total transverse energy in the detector.                                                                    |
| PRI_jet_num                   | The number of jets (integer with value of 0, 1, 2 or 3; possible larger values have been capped at 3). |
| PRI_jet_leading_pt | The transverse momentum p2x+p2y−−−−−−√ of the leading jet, that is the jet with largest transverse momentum (undefined if PRI_jet_num = 0). |
| PRI_jet_leading_eta | The pseudorapidity η of the leading jet (undefined if PRI_jet_num = 0). |
| PRI_jet_leading_phi | The azimuth angle ϕ of the leading jet (undefined if PRI_jet_num = 0). |
| PRI_jet_subleading_pt | The transverse momentum p2x+p2y−−−−−−√ of the leading jet, that is, the jet with second largest transverse momentum (undefined if PRI_jet_num ≤ 1). |
| PRI_jet_subleading_eta | The pseudorapidity η of the subleading jet (undefined if PRI_jet_num ≤ 1). |
| PRI_jet_subleading_phi | The azimuth angle ϕ of the subleading jet (undefined if PRI_jet_num ≤ 1). |
| PRI_jet_all_pt | The scalar sum of the transverse momentum of all the jets of the events. |
| Weight | The event weight wi |
| Label | The event label (string) yi ∈ {s,b} (s for signal, b for background). (objetivo) |
| KaggleSet | String specifying to which Kaggle set the event belongs : ”t”:training, ”b”:public leaderboard, ”v”:private leaderboard,”u”:unused. |
| KaggleWeight | Weight normalised within each Kaggle dataset. |


Con este dataset, se quiere predecir si un evento es una señal que corresponde a la producción de un bosón de Higgs (s) o si es un evento de fondo (b). El bosón de Higgs es una partícula elemental en el Modelo Estándar de la física de partículas y fue descubierto en 2012 en el CERN.

En este caso, el objetivo es construir un modelo de aprendizaje automático que pueda distinguir entre eventos de señal y eventos de fondo utilizando las características proporcionadas en el dataset. La variable objetivo es la columna "Label", que contiene dos posibles valores: 's' para la señal y 'b' para el fondo.

In [15]:
df_atlas

EventId  DER_mass_MMC  DER_mass_transverse_met_lep  DER_mass_vis  \
0        100000       138.470                       51.655        97.827   
1        100001       160.937                       68.768       103.235   
2        100002      -999.000                      162.172       125.953   
3        100003       143.905                       81.417        80.943   
4        100004       175.864                       16.915       134.805   
...         ...           ...                          ...           ...   
818233   918233       105.668                       46.443        60.048   
818234   918234        99.294                       30.097        62.713   
818235   918235       108.497                        9.837        65.149   
818236   918236        96.711                       20.006        66.942   
818237   918237        92.373                       80.109        77.619   

        DER_pt_h  DER_deltaeta_jet_jet  DER_mass_jet_jet  DER_prodeta_jet_jet  \
0         27.980                 0.910           124.711                2.666   
1         48.146              -999.000          -999.000             -999.000   
2         35.635              -999.000          -999.000             -999.000   
3          0.414              -999.000          -999.000             -999.000   
4         16.405              -999.000          -999.000             -999.000   
...          ...                   ...               ...                  ...   
818233   156.191                 0.403            47.746                0.936   
818234    65.861                 3.312           471.319               -2.611   
818235    18.006              -999.000          -999.000             -999.000   
818236    29.761              -999.000          -999.000             -999.000   
818237     3.984              -999.000          -999.000             -999.000   

        DER_deltar_tau_lep  DER_pt_tot  ...  PRI_jet_leading_eta  \
0                    3.064      41.928  ...                2.150   
1                    3.473       2.078  ...                0.725   
2                    3.148       9.336  ...                2.053   
3                    3.310       0.414  ...             -999.000   
4                    3.891      16.405  ...             -999.000   
...                    ...         ...  ...                  ...   
818233               1.279       6.133  ...                1.190   
818234               2.294       2.889  ...                1.293   
818235               2.742      18.006  ...             -999.000   
818236               2.479       2.739  ...                1.460   
818237               2.486       3.984  ...             -999.000   

        PRI_jet_leading_phi  PRI_jet_subleading_pt  PRI_jet_subleading_eta  \
0                     0.444                 46.062                   1.240   
1                     1.158               -999.000                -999.000   
2                    -2.028               -999.000                -999.000   
3                  -999.000               -999.000                -999.000   
4                  -999.000               -999.000                -999.000   
...                     ...                    ...                     ...   
818233               -0.766                 41.791                   0.787   
818234               -0.868                 70.158                  -2.018   
818235             -999.000               -999.000                -999.000   
818236                2.637               -999.000                -999.000   
818237             -999.000               -999.000                -999.000   

        PRI_jet_subleading_phi  PRI_jet_all_pt    Weight  Label  KaggleSet  \
0                       -2.475         113.497  0.000814      s          t   
1                     -999.000          46.226  0.681042      b          t   
2                     -999.000          44.251  0.715742      b          t   
3                     -999.000          -0.000  1.660654


Primera opción: conjunto de datos 1 (ATLAS Higgs Boson Dataset)

1-Fuente de datos:

R: CERN Open Data Portal - ATLAS Higgs Boson Dataset

2-Breve descripción de datos:

R: El conjunto de datos contiene eventos simulados del colisionador de partículas LHC del experimento ATLAS, que muestra qué sucedería si un bosón de Higgs se descompone en dos tauones.

3-¿Cuál es el objetivo? 

R: El objetivo es identificar eventos en los que el bosón de Higgs se descompone en dos tauones a partir de las características proporcionadas en el conjunto de datos.

4-¿Este es un problema de clasificación o regresión?

R: Es un problema de clasificación, ya que el objetivo es clasificar si un evento es una señal (descomposición del bosón de Higgs en dos tauones) o un fondo (proceso diferente).

5-¿Cuántas características hay? 

R: Hay 30 características en el conjunto de datos.

6-¿Cuántas filas de datos hay? 

R: Hay alrededor de 818,000 filas de datos.

7-¿Qué retos, si es que hay alguno, prevén en la limpieza, exploración o modelización con este conjunto de datos?

R: Algunos de los retos incluyen: gran cantidad de datos, características con rangos y escalas diferentes, y el desequilibrio entre eventos de señal y fondo.

https://archive.ics.uci.edu/ml/datasets/Forest+Fires  
Forest Fires Data Set


| Variable | Descripción |
| --- | --- |
| X | x-axis spatial coordinate within the Montesinho park map: 1 to 9 |
| Y | y-axis spatial coordinate within the Montesinho park map: 2 to 9 |
| month | month of the year: 'jan' to 'dec' |
| day | day of the week: 'mon' to 'sun' |
| FFMC | FFMC index from the FWI system: 18.7 to 96.20 |
| DMC | DMC index from the FWI system: 1.1 to 291.3 |
| DC | DC index from the FWI system: 7.9 to 860.6 |
| ISI | ISI index from the FWI system: 0.0 to 56.10 |
| temp | temperature in Celsius degrees: 2.2 to 33.30 |
| RH | relative humidity in %: 15.0 to 100 |
| wind | wind speed in km/h: 0.40 to 9.40 |
| rain | outside rain in mm/m²: 0.0 to 6.4 |
| area | the burned area of the forest (in ha): 0.00 to 1090.84 (this output variable is very skewed towards 0.0, thus it may make sense to model with the logarithm transform) |


In [16]:
df_fire=pd.read_csv('/content/drive/MyDrive/Coding_dojo/Proyecto2/forestfires.csv')

In [17]:
df_fire

X  Y month  day  FFMC    DMC     DC   ISI  temp  RH  wind  rain   area
0    7  5   mar  fri  86.2   26.2   94.3   5.1   8.2  51   6.7   0.0   0.00
1    7  4   oct  tue  90.6   35.4  669.1   6.7  18.0  33   0.9   0.0   0.00
2    7  4   oct  sat  90.6   43.7  686.9   6.7  14.6  33   1.3   0.0   0.00
3    8  6   mar  fri  91.7   33.3   77.5   9.0   8.3  97   4.0   0.2   0.00
4    8  6   mar  sun  89.3   51.3  102.2   9.6  11.4  99   1.8   0.0   0.00
..  .. ..   ...  ...   ...    ...    ...   ...   ...  ..   ...   ...    ...
512  4  3   aug  sun  81.6   56.7  665.6   1.9  27.8  32   2.7   0.0   6.44
513  2  4   aug  sun  81.6   56.7  665.6   1.9  21.9  71   5.8   0.0  54.29
514  7  4   aug  sun  81.6   56.7  665.6   1.9  21.2  70   6.7   0.0  11.16
515  1  4   aug  sat  94.4  146.0  614.7  11.3  25.6  42   4.0   0.0   0.00
516  6  3   nov  tue  79.5    3.0  106.7   1.1  11.8  31   4.5   0.0   0.00

[517 rows x 13 columns]

Segunda opción: conjunto de datos 2 (Forest Fires Dataset)

1-Fuente de datos:

R: UCI Machine Learning Repository - Forest Fires Dataset

2-Breve descripción de datos:

R: El conjunto de datos contiene información sobre incendios forestales en el Parque Natural Montesinho en Portugal, incluidas variables meteorológicas y geográficas.

3-¿Cuál es el objetivo? 

R: El objetivo es predecir el área quemada de un incendio forestal en función de las características proporcionadas en el conjunto de datos.

4-¿Este es un problema de clasificación o regresión? 

R: Es un problema de regresión, ya que el objetivo es predecir un valor continuo (área quemada en hectáreas).

5-¿Cuántas características hay?

R: Hay 12 características en el conjunto de datos.

6-¿Cuántas filas de datos hay? 

R: Hay 517 filas de datos.

7-¿Qué retos, si es que hay alguno, prevén en la limpieza, exploración o modelización con este conjunto de datos?

R: Algunos de los retos incluyen: variables categóricas como el mes y el día de la semana, variables con diferentes escalas y rangos, y la distribución sesgada de la variable objetivo (área quemada) hacia cero, lo que puede requerir transformaciones de los datos antes de la modelización.